src/backend/metal/src/command.rs

```diff
@@ -921,6 +985,21 @@ impl CommandSink {
         }
     }
 
+    fn quick_compute<'a, I>(&mut self, label: &str, commands: I)
+    where
+        I: Iterator<Item = soft::ComputeCommand<&'a soft::Own>>
+    {
+        {
+            let (mut pre, switch) = self.switch_compute();
+            assert!(switch);
```

Comment:
Why is this assertion guaranteed? It seems valid to have an encoder left open but call `{fill|copy}_buffer`, although I'm probably missing something here

Reply:
good point! my assumption was wrong here